src/supertux/gameconfig.cpp

```diff
@@ -49,14 +49,22 @@ Config::Config() :
   tux_spawn_pos(),
   edit_level(),
   locale(),
-  keyboard_config(),
-  joystick_config(),
+  keyboard_configs(),
+  joystick_configs(),
   addons(),
   developer_mode(false),
   christmas_mode(false),
   transitions_enabled(true),
   repository_url()
 {
+  assert(numberofplayers > 0);
+
+  for(int i = 0; numberofplayers > i; ++i){
```

Comment:
For consistency with most other parts in this codebase please use `for (int i = 0; i < expr; ++i)` the other style is not used very often and imo is harder to read.